docs/core-concepts.md

```diff
@@ -96,18 +192,108 @@ They typically record any external side effects they've triggered as events in t
+               │                                                          │
+               │          │                               │          F. Handle
+           B. Handle                                                   Query
+            Command       │                               │               │
+               │                                                          │
+               │          │                               │               │
+               ▼                       ┌─────────────┐                    ▼
+        ┌─────────────┐   │            │             │    │        ┌─────────────┐
+        │             │       ┌───────▶│  Projector  │             │             │
+     ┌──│  Aggregate  │   │   │        │             │    │        │Query Handler│
+     │  │             │       │        └─────────────┘             │             │
+     │  └─────────────┘   │  D. Read          │           │        └─────────────┘
+     │         │              event      E. Update                        │
+ A. Load   C. Emit        │   │          Projection       │               │
+  events    Event             │               │                       G. Read
+     │         │          │   │               │           │          Projection
+     │         ▼              │               ▼                           │
+     │  ┌─────────────┐   │   │        ┌─────────────┐    │               │
+     │  │             │       │        │             │                    │
+     └─▶│ Event Store │───┼───┘        │ Projection  │◀───┼───────────────┘
+        │             │                │             │
+        └─────────────┘   │            └─────────────┘    │
+
+                          │                               │
+
+```
 
-Note that clients may instead choose to run each ESP in their own process directly. The coordination of this is not currently provided by EventSourcery. 
+### 1. Handling a Command
 
-## Query Handlers
+A command comes into the application and is routed to a command handler. The command handler initialises an aggregate and loads up its state from events in the store. The command handler then defers to the aggregate to handle the command. It then stores any new events raised by the aggregate into the event store.
 
-The web layer will take a query and pass it to the appropriate query handler. The query handler reads from a projection into a model and returns it as JSON over the web.
+```ruby
+class AddTodoCommandHandler
+  def handle(id:, title:, description:)
+    # The repository provides access to the event store
+    repository = EventSourcery::Repository.new(
+      event_source: EventSourcery.config.event_source,
+      event_sink: EventSourcery.config.event_sink,
+    )
+
+    # Load up the aggregate from events in the store
+    aggregate = repository.load(TodoAggregate, id)
+
+    # Defer to the aggregate to execute the add command
```

Comment:
Maybe something like "# Defer to the aggregate to execute the add command and emit a new event"?